nixos/modules/services/web-apps/pretalx.nix

```diff
@@ -0,0 +1,406 @@
+              type = nullOr str;
+              default = optionalString cfg.celery.enable "redis+socket://${config.services.redis.servers.pretalx.unixSocket}?virtual_host=2";
+              defaultText = literalExpression ''
+                optionalString config.${opt.celery.enable} "redis+socket://''${config.${options.services.redis.servers}.pretalx.unixSocket}?virtual_host=2"
+              '';
+              description = mdDoc ''
+                URI to the celery broker used for the asynchronous job queue.
+              '';
+            };
+          };
+
+          redis = {
+            location = mkOption {
+              type = nullOr str;
+              default = "unix://${config.services.redis.servers.pretalx.unixSocket}?db=0";
+              defaultText = literalExpression ''
+                "unix://''${config.${options.services.redis.servers}.pretalx.unixSocket}?db=0"
+              '';
+              description = mdDoc ''
+                URI to the redis server, used to speed up locking, caching and session storage.
+              '';
+            };
+
+            session = mkOption {
+              type = bool;
+              default = true;
+              example = false;
+              description = mdDoc ''
+                Whether to use redis as the session storage.
+              '';
+            };
+          };
+
+          site = {
+            url = mkOption {
+              type = types.str;
+              example = "https://talks.example.com";
+              description = mdDoc ''
+                The base URI below which your pretalx instance will be reachable.
+              '';
+            };
+          };
+        };
+      };
+      default = {};
+      description = mdDoc ''
+        pretalx configuration as a Nix attribute set. All settings can also be passed
+        from the environment.
+
+        See <https://docs.pretalx.org/administrator/configure.html> for possible options.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # https://docs.pretalx.org/administrator/installation.html
+
+    services.postgresql = mkIf (cfg.database.createLocally && cfg.settings.database.backend == "postgresql") {
+      ensureUsers = [ {
+        name = cfg.settings.database.user;
+        ensurePermissions = {
+          "DATABASE ${cfg.settings.database.name}" = "ALL PRIVILEGES"; };
+      } ];
+      ensureDatabases = [ cfg.settings.database.name ];
+    };
+
+    services.redis.servers.pretalx.enable = true;
+
+    services.nginx = mkIf cfg.nginx.enable {
+      additionalModules = with pkgs.nginxModules; [
+        moreheaders
+      ];
+      recommendedGzipSettings = mkDefault true;
+      recommendedOptimisation = mkDefault true;
+      recommendedProxySettings = mkDefault true;
+      recommendedTlsSettings = mkDefault true;
+      virtualHosts.${cfg.nginx.domain} = {
+        # https://docs.pretalx.org/administrator/installation.html#step-7-ssl
+        extraConfig = ''
+          gzip off;
```

Comment:
```suggestion
```
I don't see a reason in the upstream docs why this should be disabled, so I would leave it out for now.